File: src/main/scala/actors5/Lector.scala
package hello.actors5

import scala.actors.Actor
import scala.actors.Actor._
import java.io._
import java.net.Socket

object Lector extends Actor {

  def act() {
    loop {
      react {
        case s:Socket =>
          val reader = new BufferedReader(new InputStreamReader(s.getInputStream))
          val cuando = new java.util.Date
          val nombre = reader.readLine()
          Escritor ! Saluda(nombre, cuando, s)
      }
    }
  }
}

File: src/main/java/hola/Lector.java
package hola;
import java.net.*;
import java.io.*;

public class Lector implements Runnable {

  private final Socket s;
  private Lector(Socket socket) {
    s = socket;
  }
  public void run() {
    try {
      BufferedReader input = new BufferedReader(new InputStreamReader(s.getInputStream()));
      String nombre = input.readLine();
      //Echemosle la bronca de hacer otro hilo al escritor
      Escritor.escribe(nombre, s);
    } catch (IOException ex) {
      //Algo malo ocurrio que hacemos?
    }
  }

  public static void leer(Socket s) {
    new Thread(new Lector(s)).start();
  }

}
